### DISCUSSION TITLE
combining @default(now()) and @unique() on same key

### DISCUSSION THREAD

**Discussion by lskuper** [prisma/prisma#4903] [Q&A] [answered]
Hi there,

Can combining `@default(now())` and `@unique()` on one column lead to errors when creating objects frequently/in parallel?

schema:
```prisma
 model Post { 
   id        String   @id @default(uuid()) 
   createdAt DateTime @default(now())  @unique()
 }
```
scenario: multiple threads calling `prismaClient.post.create({});` at the same time.
Or does @default(now()) guarantee that the unique constraint is fulfilled? (assume I never pass createdAt manually and give always prisma the power to create the value).

Cheers

**Comment by ryands17** [comment, marked as answer]
Hey @lskuper 👋 
Yes this will cause a problem when you create 2 or more records at the same time as the timestamp attached will be the same and you would get a unique constraint violation error.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
